lib/controllers/users.js

```diff
@@ -0,0 +1,57 @@
+const { Router } = require('express');
+const UserService = require('../services/UserService');
+const { User } = require('../models/User');
+const authenticate = require('../middleware/authenticate');
+const authorize = require('../middleware/authorize');
+
+const ONE_DAY_IN_MS = 1000 * 60 * 60 * 24;
+
+module.exports = Router()
+  .post('/', async (req, res, next) => {
+    try {
+      const user = await UserService.create(req.body);
+      res.json(user);
+    } catch (error) {
+      next(error);
+    }
+  })
+  .post('/sessions', async (req, res, next) => {
+    try {
+      const token = await UserService.signIn(req.body);
+      res
+        .cookie(process.env.COOKIE_NAME, token, {
+          httpOnly: true,
+          maxAge: ONE_DAY_IN_MS,
+        })
+        .json({ message: 'WELCOME USER, TO YOUR DOOOM!' });
+    } catch (e) {
+      next(e);
+    }
+  })
+
+  .get('/secrets', [authenticate, authorize], async (req, res, next) => {
```

Comment:
what's the difference between hitting ```/secrets``` and ```/users/secrets``` (like you're doing here)? this route isn't doing anything, it shouldn't exist. i could imagine a world where there were secrets associated with certain users, a secret for user 1, secret for user 2, and you use authorize middleware to only allow user 1 to see user 1s secrets, but that doesn't have anything to do with the actual data you're working with in this one — from the rubric:
"Your first task is to build an API and database schema for storing user emails & (hashed) passwords, as well as endpoints for creating and listing notes that can only be accessed by authenticated users...
Logged in users can view top secrets by visiting /api/v1/secrets
Logged in users can create new secrets by POSTing to /api/v1/secrets"
/users — the table and how we interact with our table for the users who are allowed to view the secrets
/secrets — the table that contains a list of secrets, it should be restricted to only be seen by people who are in our users table, i.e. people who have a ticket to the concert, so we use authenticate middleware